apps/src/weblab/WebLab.js

```diff
@@ -163,6 +163,12 @@ WebLab.prototype.init = function (config) {
       onPuzzleComplete: this.onFinish.bind(this),
       unsubmitUrl: this.level.unsubmitUrl
     });
+
+    // The project template workspace icon callout appears too far to the left
+    // if we don't delay this slightly (currently 0.1 seconds)
+    setTimeout(() => {
+      this.studioApp_.initProjectTemplateWorkspaceIconCallout();
+    }, 100);
```

Comment:
This is interesting.  Any idea what changes in those 100ms?  If this is tied to bramble load time I'm worried that 100ms won't be enough on slow connections, or for users that haven't cached any of that content yet.  I'm also curious if this is still a problem if we tie this call to the componentDidMount event (I kind of assume it would be, since it's still finding the DOM in the earlier call).

Reply:
It is already in the `onMount` of the `WebLabView` component. I tried for a while to figure out what was going on. I had a fix that used our "snap callouts" function that is used in droplet/blockly quite a bit (that basically fixes up the positions). But it looked strange to make it appear and then move it. There is something happening asynchronously in the React world (I believe) after the `componentDidMount` is complete. I tried hooking something to `componentDidUpdate`, but that doesn't get called at all for the initial load.
I'm open to suggestions, but figure that for a callout only on weblab, it probably wasn't worth going through a deeper multi-hour investigation. And I sort of like having a delay where the page loads and then the callout pops. Increasing the delay would be fine for the user experience in this case, but I can't prove what delay in "enough" since I don't really understand what is changing during that brief period of time...

Reply:
Sounds good.  I wish we had a better understanding of this, but it doesn't need to block anything. :+1: